src/alert-prelude.c

```diff
@@ -696,10 +700,16 @@ static void PacketToDataProtoHTTP(const Packet *p, const PacketAlert *pa, idmef_
  */
 static void PacketToDataProtoHTTP2(const Packet *p, const PacketAlert *pa, idmef_alert_t *alert)
 {
-    void *http2_state = FlowGetAppState(f);
+    void *http2_state = FlowGetAppState(p->flow);
+    json_error_t error;
+    JsonBuilder *jsb;
+
     if (http2_state) {
         void *tx_ptr = rs_http2_state_get_tx(http2_state, pa->tx_id);
-        json_t *js = rs_http2_log_json(tx_ptr);
+        jsb = jb_new_object();
+        rs_http2_log_json(tx_ptr, jsb);
+        jb_close(jsb);
+        json_t *js = json_loadb((char *)jb_ptr(jsb), jb_len(jsb), 0, &error);
```

Comment:
I don't like this approach. If the prelude output is essentially going to parse the eve-json string, why not move it into an external process? It seems that eve contains enough info for that, and if anything is missing we can consider adding it.

Reply:
Hello,

Not essentially, just for extra data. If there is some already processed data, why don't add them as additional data?

If you really don't want this, I see two options :
* In the http code, split functions so Prelude can call them and get data without parsing JSON
* Remove these additionals data for the Prelude connector

Regards

Reply:
I think for 6 (`master-6.0.x`) we can accept this approach, but for the suricata 7 side (`master`) I would like to get rid of it. Then there are several options: 1) see if it fits our new output plugin support 2) make it a completely separate tool that post processes EVE.

It seems that there aren't many users anyway. The only report we've had was on the debian package not building. But no actual users complaining yet.

Reply:
Agreed on this approach for 6.. But I think we should remove it from master sooner than later.  Might get a plugin going, or better yet, seems like some log watching agent is the best for this kind of stuff, at least in my opinion.